engine/grid.py

```diff
@@ -253,44 +265,51 @@ def area(x1, y1, x2, y2, x3, y3):
 
         # Check if sum of A1, A2 and A3
         # is same as A
-        if (A == A1 + A2 + A3):
-            return True
+        return A == (A1 + A2 + A3)
+
+    def activate_line(self, row, col, n, isHorizontal):
+        """
+        Activate a single horizontal line of length n starting at row, col.
+        If isHorizontal is True, the line is from (row, col) to (row, col + n). 
+        Otherwise, the line is from (row, col) to (row + n, col). 
+        """
+        # Note: rows are y-position and columns are x-position
+        if isHorizontal:
+            for i in range(n):
+                self.nodes[col+i][row].is_active = True
         else:
-            return False
+            for i in range(n):
+                self.nodes[col][row+i].is_active = True
 
+    ##Activate triangle based on three points
     def activate_triangle(self, x1, y1, x2, y2, x3, y3):
         """
-        Activates all the nodes in a traingle based on three points.
```

Comment:
why change this spec? I think it would be nice to be specific. If the problem is redundancy (x color of 1 point, x color of 2nd point), can do currying and uncurrying -> param takes in 3 points instead of 3 x coor and 3 y coor. Can then specify that the points are vertices of the triangle

Reply:
This is a good point, I will change the inputs to points